operator/v1/types_network.go

```diff
@@ -425,6 +425,88 @@ type OVNKubernetesConfig struct {
 	// egressIPConfig holds the configuration for EgressIP options.
 	// +optional
 	EgressIPConfig EgressIPConfig `json:"egressIPConfig,omitempty"`
+	// ipv4 allows users to configure IP settings for IPv4 connections. When ommitted,
+	// this means no opinions and the default configuration is used. Check individual
+	// fields within ipv4 for details of default values.
+	// +optional
+	IPv4 *IPv4OVNKubernetesConfig `json:"ipv4,omitempty"`
+	// ipv6 allows users to configure IP settings for IPv6 connections. When ommitted,
+	// this means no opinions and the default configuration is used. Check individual
+	// fields within ipv4 for details of default values.
+	// +optional
+	IPv6 *IPv6OVNKubernetesConfig `json:"ipv6,omitempty"`
+}
+
+type IPv4OVNKubernetesConfig struct {
+	// internalTransitSwitchSubnet is a v4 subnet in IPV4 CIDR format used internally
+	// by OVN-Kubernetes for the distributed transit switch in the OVN Interconnect
+	// architecture that connects the cluster routers on each node together to enable
+	// east west traffic. The subnet chosen should not overlap with other networks
```

Comment:
does anything validate this?

Reply:
yes https://github.com/openshift/cluster-network-operator/pull/2318/commits/4b797b1988c2c50ebbc875896392c301c702485e#diff-e68f94c90e1c01972187483984b3da57967f404e4e2a85a48433af6ca9d34c8c we will check if any of the configurable subnets overlap here in the CNO when this merges